### PR TITLE
Soar-8241: Help.md Validator Update

### PR DESCRIPTION
**Pull request by aholland-r7**
### Description

Updating the validators for help.md to match the needs of plugin spec v3
Required:
  Input examples
  Output examples
  Title (across the board)
  Description (definitely components and inputs, not outputs)
  Links (at least a vendor link)
  Version History
  Key features

Non-Required:
  Help section (per component)
  Troubleshooting (per component)
  Requirements

## PR Requirements

Developers, verify you have completed the following items by checking them off:

- [x] Unit tests updated for any new or updated code
- [x] Version bumped within setup.py
- [x] Changelog entry added

### CODE REVIEW

File: unit_test/plugin_examples/good_plugin/help.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ Example input:
 
 |Name|Type|Required|Description|
```
Comment (tcheng-r7) [inline unit_test/plugin_examples/good_plugin/help.md:47]: Looks like you're adding a column with examples but the title row didn't have a Example title added to it?

File: unit_test/plugin_examples/good_plugin_with_task/help.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ Example input:
 
 |Name|Type|Required|Description|
```
Comment (tcheng-r7) [inline unit_test/plugin_examples/good_plugin_with_task/help.md:48]: Same thing here

**Review by tcheng-r7 — APPROVED** [review]

File: icon_validator/rules/plugin_validators/help_input_output_validator.py [hunk 3]
```diff
@@ -83,7 +83,7 @@ def get_spec_input(input_content: dict) -> list:
             required = input_content.get(k).get("required")
             description = input_content.get(k).get("description")
             enum = input_content.get(k).get("enum", None)
-            example = input_content.get(k).get("example", None)
```
Comment (tcheng-r7) [inline icon_validator/rules/plugin_validators/help_input_output_validator.py:86]: What was the reasoning behind removing the default of None here?

Comment (aholland-r7) [reply icon_validator/rules/plugin_validators/help_input_output_validator.py:86]: I want to make it so you have to have added an example.

Comment (tcheng-r7) [reply icon_validator/rules/plugin_validators/help_input_output_validator.py:86]: Ah right, that makes sense

Comment (mrinehart-r7) [reply icon_validator/rules/plugin_validators/help_input_output_validator.py:86]: `None` is also the default argument by default for `.get()` :)

File: icon_validator/rules/plugin_validators/help_validator.py [hunk 4]
```diff
@@ -65,6 +65,19 @@ def remove_example_output(help_content):
...
+    def validate_required_content(help_raw: str):
+        help_content = help_raw.replace("\n", '')
+        pattern1 = "# Key Features(.*?)# Requirements"
+        pattern2 = "# Links(.*?)## References"
+        key_features = re.search(pattern1, help_content).group(1)
+        if "*" not in key_features:
+            raise ValidationException(f"Help section is missing list of Key Features in help.md, "
+                                      f"must include at least one feature")
+        links = re.search(pattern2, help_content).group(1)
+        if "https:" not in links:
```
Comment (mrinehart-r7) [inline icon_validator/rules/plugin_validators/help_validator.py:78]: What about http?

**Review by tcheng-r7 — APPROVED** [review]

**Review by blaxminarayan-r7 — APPROVED** [review]

**Pull request merged**